pliers/filters/text.py

```diff
@@ -49,8 +55,80 @@ def __init__(self, stemmer='porter', *args, **kwargs):
             raise ValueError("stemmer must be either a valid string, or an "
                              "instance of class StemmerI.")
         self.stemmer = stemmer
+        self.tokenize = tokenize
         super(WordStemmingFilter, self).__init__()
 
     def _filter(self, stim):
-        stemmed = self.stemmer.stem(stim.text)
+        if self.tokenize:
+            tokens = stim.text.split()
+            stemmed = ' '.join([self.stemmer.stem(tok) for tok in tokens])
+        else:
+            stemmed = self.stemmer.stem(stim.text)
         return TextStim(stim.filename, stemmed, stim.onset, stim.duration)
+
+
+class TokenizingFilter(TextFilter):
+
+    ''' Tokenizes a TextStim into several word TextStims.
+    Args:
+        tokenizer (nltk Tokenizer): a nltk Tokenizer to tokenize
+            with. Will use the word_tokenize method if none is specified.
+    '''
+
+    _log_attributes = ('tokenizer',)
+
+    def __init__(self, tokenizer=None):
+        self.tokenizer = tokenizer
+        super(TokenizingFilter, self).__init__()
+
+    def _filter(self, stim):
+        if self.tokenizer:
+            tokens = self.tokenizer.tokenize(stim.text)
+        else:
+            tokens = word_tokenize(stim.text)
+        stims = [TextStim(stim.filename, token, stim.onset, stim.duration)
+                 for token in tokens]
+        return stims
+
+
+class TokenRemovalFilter(TextFilter):
+
+    ''' Removes tokens (e.g. stopwords, common words, punctuation) from a
+    TextStim.
+
+    Args:
+        tokens (list): a list of tokens (strings) to remove from a
+            TextStim. Will use nltk's default stopword list if none is
+            specified.
+        language (str): if using the default nltk stopwords, specifies
+            which language from which to use stopwords.
+    '''
+
+    _log_attributes = ('tokens', 'language')
+
+    def __init__(self, tokens=None, language='english'):
+        self.language = language
+        if tokens:
+            self.tokens = set(tokens)
+        else:
+            try:
+                nltk.data.find('corpora/stopwords')
```

Comment:
I could be wrong, but I think I wrote a decorator somewhere that allows you to specify what corpora you need, instead of having to specify it inside the function call.

Reply:
Yeah I didn't really want to use it since its not explicitly necessary for `_extract` to run, only as a default. Moreover, as a default, it seemed to make more sense to force the download manually.